assets/js/script.js

```diff
@@ -175,16 +177,146 @@ function edamamFetch() {
+    dropdown.appendChild(option);
+  });
+}
+
+// Populate dropdown on page load
+document.addEventListener('DOMContentLoaded', function() {
+  // Load ingredients from local storage if available
+  spoonIngredientArray = JSON.parse(localStorage.getItem("spoonIngredientArray")) || [];
+  edamamIngredientArray = JSON.parse(localStorage.getItem("edamamIngredientArray")) || [];
+
+  updateDropdown();
+});
+
+
+
+
+// Attaches a change event listener to the dropdown
+document.getElementById('ingredientDropdown').addEventListener('change', function() {
+  // Get the selected value
+  var selectedIngredient = this.value;
+
+  // Get the current content of the textarea
+  var currentText = document.getElementById('spoonIngredientInput').value;
+
+  // Append the selected ingredient to the textarea content
+  // You can modify the format of how it's appended as needed
+  if (currentText.length > 0) {
+      currentText += ', ';
+  }
+  
+  currentText += selectedIngredient;
+
+  // Update the textarea with the new content
+  document.getElementById('spoonIngredientInput').value = currentText;
+});
+
+document.getElementById('ingredientShopDropdown').addEventListener('change', function() {
+  // Get the selected value
+  var selectedIngredient = this.value;
+
+  // Get the current content of the Edamam textarea
+  var currentText = document.getElementById('edamamIngredientInput').value;
+
+  // Append the selected ingredient to the textarea content
+  if (currentText.length > 0) {
+      currentText += ', ';
+  }
+  currentText += selectedIngredient;
+
+  // Update the textarea with the new content
+  document.getElementById('edamamIngredientInput').value = currentText;
+});
+
+function updateEdamamDropdown() {
+  const dropdown = document.getElementById('ingredientShopDropdown');
+  dropdown.innerHTML = ''; // Clear existing options
+
+  edamamIngredientArray.forEach(ingredient => {
+    const option = document.createElement('option');
+    option.value = ingredient;
+    option.textContent = ingredient;
+    dropdown.appendChild(option);
+  });
+}
+
+function populateEdamamDropdown() {
+  const dropdown = document.getElementById('ingredientShopDropdown');
+  const ingredients = JSON.parse(localStorage.getItem('edamamIngredientArray'));
+
+  if (ingredients) {
+      dropdown.innerHTML = ''; // Clear existing options
+      ingredients.forEach(ingredient => {
+          const option = document.createElement('option');
+          option.value = ingredient;
+          option.textContent = ingredient;
+          dropdown.appendChild(option);
+      });
+  }
+}
+
+
+
 $("#edamamSearchBtn").on("click", function () {
-  // Fills the empty ingredient arrays with the users input.
-  edamamIngredientArray = $("#edamamIngredientInput").val().split(", ");
-  // Calls the Fetch function.
+  edamamIngredientArray = JSON.parse(localStorage.getItem("edamamIngredientArray")) || [];
+  let newIngredients = $("#edamamIngredientInput").val().split(", ");
```

Comment:
I think this and the other code around this is causing the array to always be added to and never reset to blank.